iceoryx_examples/callbacks/README.md

```diff
@@ -184,6 +194,113 @@ Afterward, we reset both caches to start fresh again.
     }
 ```
 
+### ice_callbacks_listener_as_class_member.cpp
+
+This example is identical to the [ice_callbacks_subscriber.cpp](#ice_callbacks_subscriber.cpp) 
+one except that we left out the cyclic heartbeat trigger. The key difference is that 
+the listener is now a class member and in every callback we would like to change 
+some member variables. To do this we require an additional pointer to the object 
+since the listener requires c function references which do not allow the usage
+of lambdas with capturing. Here we can use the userType feature which allows us 
+to provide the this pointer as additional argument to the callback.
+
+The main function is now pretty short, we instantiate our object of type `CounterClass`
+and call `waitForControlC` which uses the `shutdownSemaphore` like in the 
+previous example to wait for the control c event from the user.
+```cpp
+auto signalIntGuard = iox::posix::registerSignalHandler(iox::posix::Signal::INT, sigHandler);
+auto signalTermGuard = iox::posix::registerSignalHandler(iox::posix::Signal::TERM, sigHandler);
+
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
+
+CounterClass counterClass;
+
+counterClass.waitForControlC();
+```
+
+Our `CounterClass` has the following members:
+```cpp
+class CounterClass {
+    //...
+    iox::popo::Subscriber<CounterTopic> m_subscriberLeft;
+    iox::popo::Subscriber<CounterTopic> m_subscriberRight;
+    iox::cxx::optional<CounterTopic> m_leftCache;
+    iox::cxx::optional<CounterTopic> m_rightCache;
+    iox::popo::Listener m_listener;
+};
+```
+
+And their purposes are the same as in the previous example. In the constructor
+we initialize the two subscribers and attach them to our listener. But now we 
+add an additional parameter in the `iox::popo::createEventCallback` the 
```

Comment:
the __n__

Reply:
I'm dumb, see https://github.com/eclipse-iceoryx/iceoryx/pull/715#discussion_r612400118